### DISCUSSION TITLE
Only get hash value using md5sum (without filename)

### DISCUSSION THREAD

**Discussion by otakutyrant** [nushell/nushell#10614] [Q&A] [answered]
Have a look at this [question](https://stackoverflow.com/questions/3679296/only-get-hash-value-using-md5sum-without-filename) please.

I have tried `md5sum file | get 0` or `md5sum file | str split`, but both did not work. Is there any viable and more elegant way in Nushell rather than using traditional awk or cut command?

**Comment by hustcer** [comment, marked as answer]
```nu
open -r file | hash md5
```

**Reply by otakutyrant** [reply]
Thank you.

**Comment by ghost** [comment]
hustcer's answer is probably ideal in this case, but for learning nu here are a few other methods:

> I have tried `md5sum file | get 0` or `md5sum file | str split`, but both did not work.

you have to combind them - first split it, then select the part you want (also `str split` is `split row`):
```nu
md5sum .bashrc | split row ' ' | get 0
```

since md5 hashes are always 32 characters long it is also possible to use
```nu
md5sum .bashrc | str substring 0..32
```

md5sum can hash multiple files in a single command (which is nice for `md5sum *.txt` and similar). we can parse this with
```nu
md5sum .bashrc .profile | lines | parse '{hash}  {file}'
# lines splits it at newline characters into a list of strings
# parse parses a string and fills in the placeholders we placed (`{hash}` and `{file}`)
# NOTE: md5sum uses 2 spaces as seperator -> we also have to use 2 spaces

# result: [{hash: a5474cd2d3f9e25a46e8191795ee863b, file: .bashrc}, {hash: fbc3c9841beee1e2b8f64d6ae8d2b13c, file: .profile}]
```

**Reply by panicbit** [reply]
> md5sum uses 2 spaces as seperator

Note that this is not quite accurate. From the md5sum man page:

> The default mode is to print a line with: checksum, a space, a character indicating input mode ('*' for binary, ' ' for text or where binary is insignificant), and name for each FILE.

**Reply by otakutyrant** [reply]
I didn't realize that there are so many useful internal commands like `hash` and `split`. Since then I will check internal commands at first, thank you.
